src/Pelagos/Bundle/AppBundle/Resources/public/js/data-discovery.js

```diff
@@ -99,8 +106,213 @@ function resizeLeftRight() {
     $("#right").height(rh);
 }
 
+function formatBytes(bytes,decimals) {
+    if(bytes == 0) return '0 Bytes';
+    var k = 1024,
+        dm = decimals <= 0 ? 0 : decimals || 2,
+        sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'],
+        i = Math.floor(Math.log(bytes) / Math.log(k));
+    return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
+}
+
+function getActiveTabIndex() {
+    var activeTabIndex = parseInt($("#tabs").tabs("option","active"));
+    if (isNaN(activeTabIndex)) {
+        activeTabIndex = 0;
+    }
+    return activeTabIndex;
+}
+
+//this function gets data from the server-side and parse it to a buffer array
+function loadData(by, id) {
+    var activeTabIndex = getActiveTabIndex();
+    geo_filter = "";
+    if (!by && !id) {
+        by = $("#by-input").val()
+        id = $("#id-input").val()
+    }
+
+    if (trees["tree"].geo_filter) {
+        geo_filter = trees["tree"].geo_filter;
+    }
+    //this gets the data from the search in a bulk based on bulkSize and currentIndex (similar to index for pagination)
+    return jqXHR = $.ajax({
+        "url": Routing.generate("pelagos_app_ui_datadiscovery_search", { "id": id }),
+        "data": {
+            "filter": jQuery("#filter-applied").val(),
+            "by": by,
+            "id": id,
+            "geo_filter": geo_filter,
+            "active_tab_index": activeTabIndex,
+            "current_index": currentIndex,
+            "bulk_size": bulkSize
+        },
+        "dataType": 'json',
+        "success": function(response) {
+            buffer = JSON.parse(response);
+
+        },
+        "error": function(jqXHR, textStatus, errorThrown) {
+            console.log("Fail: " + textStatus + " " + errorThrown + jqXHR.getResponseHeader());
+        }
+
+    }).done(function(){ //addRows to the table
+       addRows();
+    });
+};
+
+//this iterates new bulk of data and render new rows
+function addRows() {
+    var activeTabIndex = getActiveTabIndex();
+    var activeTabSelector = "table.datasets[tabIndex=" + activeTabIndex + "]";
+
+    buffer.forEach(dataset => {
+        var data = dataset["_source"];
+        var row = $(activeTabSelector + " tr:eq(" + currentIndex + ")");
+        row.attr("udi", data["udi"]);
+        row.attr("datasetid", dataset["_id"]);
+
+        var rowContent = createRow(data, row);
+        row.html(rowContent);
+        row.show();
+        currentIndex++;
+        //add the data to the data array
+        datasetList[activeTabIndex][dataset["udi"]] = dataset;
```

Comment:
dataset["udi"] seems to be undefined. I think it is data["udi"] just like LIne no.172 ?

Reply:
good catch!!!!!!